Frameworks/CoreGraphics/D2DUtility.mm

```diff
@@ -14,14 +14,28 @@
 //
 //******************************************************************************
 
-#import "D2DWrapper.h"
+#import "D2DUtility.h"
 #import <Starboard.h>
 
 using namespace Microsoft::WRL;
 
 // Helper for creating a D2DFactory
 ComPtr<ID2D1Factory> _CreateD2DFactoryInstance() {
-    ComPtr<ID2D1Factory> d2dFactory;
-    THROW_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, __uuidof(ID2D1Factory), &d2dFactory));
-    return d2dFactory;
+  ComPtr<ID2D1Factory> d2dFactory;
+  THROW_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED,
+                                    __uuidof(ID2D1Factory), &d2dFactory));
+  return d2dFactory;
+}
+
+static ComPtr<IWICImagingFactory> __GetWICFactory() {
+  ComPtr<IWICImagingFactory> wicFactory;
+  THROW_IF_FAILED(CoCreateInstance(CLSID_WICImagingFactory, nullptr,
+                                   CLSCTX_INPROC_SERVER,
+                                   IID_PPV_ARGS(&wicFactory)));
+  return wicFactory;
+}
+
+ComPtr<IWICImagingFactory> _GetWICFactory() {
+  static ComPtr<IWICImagingFactory> s_WICFactory = __GetWICFactory();
```

Comment:
nit: just create in a lambda rather than having a static private method that's just called in here